packages/components/molecules/f-media-element/src/components/MediaElement.vue

```diff
@@ -1,33 +1,283 @@
+                imageAlignClass
+            ]">
+            <img
+                :class="[$style['c-mediaElement-img'], 'c-mediaElement-img--override']"
+                :src="imageUrl"
+                :alt="altText">
+        </div>
     </div>
 </template>
 
 <script>
+import { ALIGN, FONT_SIZE } from '../constants';
 
 export default {
+    props: {
+        title: {
+            type: String,
+            required: true
+        },
+        text: {
+            type: String,
+            required: true
+        },
+        imageUrl:  {
+            type: String,
+            required: true
+        },
+        stacked: {
+            type: Boolean,
+            default: false
+        },
+        reverse: {
+            type: Boolean,
+            default: false
+        },
+        contentAlign: {
+            type: String,
+            default: ALIGN.LEFT
+        },
+        imageAlign: {
+            type: String,
+            default: ALIGN.LEFT
+        },
+        textSize: {
+            type: String,
+            default: 'md'
+        },
+        altText: {
+            type: String,
+            default: 'Media Element Image'
+        }
+    },
     data () {
         return {
             componentName: 'MediaElement'
         };
+    },
+    computed: {
+        /**
+         * Returns the class to modify the alignment of the content based on the prop contentAlign
+         * @returns {*}
+         */
+        contentAlignClass () {
+            switch (this.contentAlign) {
+                case ALIGN.LEFT:
+                    return this.$style['c-mediaElement-content--left'];
+                case ALIGN.RIGHT:
+                    return this.$style['c-mediaElement-content--right'];
+                case ALIGN.CENTER:
+                    return this.$style['c-mediaElement-content--center'];
+                default:
+                    return this.$style['c-mediaElement-content--left'];
+            }
+        },
+        /**
+         * Returns the class to modify the alignment of the image based on the prop imageAlign
+         * @returns {*}
+         */
+        imageAlignClass () {
+            switch (this.imageAlign) {
+                case ALIGN.LEFT:
+                    return this.$style['c-mediaElement-imgWrapper--left'];
+                case ALIGN.RIGHT:
+                    return this.$style['c-mediaElement-imgWrapper--right'];
+                case ALIGN.CENTER:
+                    return this.$style['c-mediaElement-imgWrapper--center'];
+                default:
+                    return this.$style['c-mediaElement-imgWrapper--left'];
+            }
+        },
+        /**
+         * Returns the class to modify the font size of the content based on the prop textSize
+         * @returns {*}
+         */
+        fontSizeClass () {
+            switch (this.textSize) {
+                case FONT_SIZE.SM:
+                    return this.$style['c-mediaElement--fontSizeSmall'];
```

Comment:
Super NITT – as this is applied on the same level as `c-mediaElement-content`, the modifier should be `c-mediaElement-content--fontSizeSmall` etc.

Could also probably make it `c-mediaElement-content--small` `c-mediaElement-content--medium` and I think it would still make sense as a classname?

Reply:
Agree on the first point disagree with the second. As the class purely effects the font size only, I think it's best to name it as such.

Reply:
Cool – happy for you to keep it as `fontSizeSmall` etc 👍